tests/utils.py

```diff
@@ -154,43 +147,114 @@ class TestServerProcess():
 
 
   def __init__(self, log, server='simple_server.py',
-      port=None, timeout=10, popen_cwd=".",
-      extra_cmd_args=[]):
+      timeout=10, popen_cwd=".", extra_cmd_args=[]):
 
     # Create temporary log file used for logging stdout and stderr
-    # of the subprocess. In the mode "r+"" stands for reading and writing
+    # of the subprocess. In the mode "r+" stands for reading and writing
     # and "t" stands for text mode.
     self.__temp_log_file = tempfile.TemporaryFile(mode='r+t')
 
     self.server = server
-    self.port = port or random.randint(30000, 45000)
     self.__logger = log
+    try:
+      self._start_server(timeout, extra_cmd_args, popen_cwd)
+      wait_for_server('localhost', self.server, self.port, timeout)
+    except Exception as e:
+      # Clean the resources and log the server errors if any exists.
+      self.clean()
+      raise e
+
+
+
+  def _start_server(self, timeout, extra_cmd_args, popen_cwd):
+    """Start the server subprocess. Uses a retry mechanism
+    if the bind fails."""
+
+    success = False
+    retries = 0
+    elapsed = 0
+    start = time.time()
+    while not success and elapsed < timeout:
```

Comment:
I understood this outer loop should not needed anymore  -- why are we retrying if we expect the bind() to choose a free port for us in the subprocess?

The docstring above should be then changed as well: no retries

Reply:
Yes, I will remove this loop.